util/validate.go

```diff
@@ -1,19 +1,42 @@
 package util
 
 import (
-	"fmt"
 	"regexp"
 
 	"github.com/prometheus/common/model"
 )
 
-var validLabelRE = regexp.MustCompile(`^[a-zA-Z_][a-zA-Z0-9_]*$`)
+// Error see https://dave.cheney.net/2016/04/07/constant-errors
+type Error string
+
+func (e Error) Error() string { return string(e) }
+
+// Errors returned by ValidateSample
+const (
+	ErrMissingMetricName = Error("sample missing metric name")
+	ErrInvalidMetricName = Error("sample invalid metric name")
```

Comment:
Probably useful to put "name" in quotes here, to indicate that we are missing the metric called 'name'.